Objects/StrategyCardAutomator.ttslua

```diff
@@ -1623,10 +1624,20 @@ function researchTech(color, cardName)
 	-- get technology deck
 	local deckName = 'Technology Cards'
 	local techDeck = false
-	for _, object in ipairs(getAllObjects()) do
-		if (object.tag == 'Deck') and (object.getName() == deckName) and (_zoneHelper.zoneFromPosition(object.getPosition()) == color) then
-			techDeck = object
-			break
+	local techBoard = false
+	local techCardPositionsOnTable = {}
+	local techType = _technologyHelper.getTechType(cardName)
+	for _,object in ipairs(getAllObjects()) do
+		if (color == _zoneHelper.zoneFromPosition(object.getPosition())) then
```

Comment:
I don't know how expensive cross-objects calls are; for this kind of thing I like to:
1. get the set of tech card names in a single call (e.g. _technologyHelper.techNameSet()).
2. in the loop test the name against  build up guidToPosition for tech.
3. do the zone check for boards and decks only after the name matches so there would only be ~12 in a 6 player game.

Then after the table scan can do the bulk zonesFromPositions on the tech cards, finally extracting just the ones in zone.

Reply:
I didn't realize, that cross-object calls might be an issue. In that case this makes perfect sense.